proto/internal/temporal/server/api/persistence/v1/executions.proto

```diff
@@ -254,6 +254,9 @@ message WorkflowExecutionInfo {
 
     // The workflow has been reset.
     bool workflow_was_reset = 96;
```

Comment:
I think this needs to be renamed or removed. Since existence of `reset_run_id` indicates that the current run was reset. Will fix it in a separate PR.

Reply:
Ah, thanks! Yours is better. I don't think this is used in any released version so should be safe to remove.

Reply:
Hmm... just have to make sure that `reset_run_id` is set before we trigger workflow close callbacks.

Reply:
I think that flag is still needed. Going through the code, it looks like @justinp-tt did it to track if a workflow was terminated due to reset so that some post termination behavior can be altered.

My change records this new run only in the base run (product requirement). If base run is different from the current run, we wont have anything to indicate that the current run was terminated due to reset.
I'll just rename it to `was_terminated_due_to_reset` to make it less ambiguous.

Reply:
hmmm the callback of the current run can be different from the callback of the base run right? So we should still trigger the callback. I guess the only reset case it can be skipped is when base == current. And logically we don't need any field in executionInfo which is persisted for making this skipping logic to work.

Reply:
Talked to @yycptt offline, there's a bug in the current implementation. I wasn't aware that you can reset to the non-current run ID when reviewing @justinp-tt's PR.

We'll need to change the behavior to trigger callbacks if the current workflow was reset using a different run as the base.

Reply:
@gow can you confirm the behavior of `reset_run_id`? Would we be able to use this field when considering whether to trigger the close callbacks?

Reply:
I'm not completely familiar with the Nexus close callback scenarios. I'm assuming it's important to know if a run was terminated due to reset so that you can prevent the callbacks from running. Instead you want to callbacks to be copied over to the new run and invoked when the new run closes.

If this is the case then `reset_run_id` would not be helpful since this is recorded only on the base runs. If the base run is different from the current run, then current run will simply be terminated. [Looks like we have `ndc.IsTerminatedByResetter(event)`](https://github.com/temporalio/temporal/blob/73f4e2047d19dc906c7cc0ec6968c0cd331cc746/service/history/ndc/workflow_resetter.go#L871). Maybe that can be used?

Reply:
I think we just need to check if `reset_run_id` is empty or not in `processCloseCallbacks`. Trigger the callbacks only when it's empty (and not continue as new/cron/retry). 
If base == current, then the field will be set and we won't trigger the callback. Otherwise, the field won't be set, and callback should be triggered. The field may also be set after the workflow is closed, but we only care about the transaction that closes the current run.

Reply:
My understanding was that the callbacks should not be triggered when a run is terminated due to reset. i.e even when base != current, we want to prevent the callbacks. If that assumption is incorrect then yes checking `reset_run_id` will work.